### ISSUE TITLE
[serve] worker died - trying to pass python handle to actor 

### ISSUE THREAD

**Issue by jasonyeoh87** [ray-project/ray#10188]
### What is the problem?

I am trying to set up a batch ray serve that accepts requests from more than one actors. I'm doing this by passing the Ray Serve python handle to the actors as an argument when spawning the process. Not sure if this is supported or if there is a better way to do this.

The error I got was:
  E0819 09:14:36.353615 12152 13332 task_manager.cc:323] Task failed: IOError: cancelling all pending tasks of dead actor: Type=ACTOR_TASK, Language=PYTHON, Resources: {},
   ray.exceptions.RayActorError: The actor died unexpectedly before finishing this task.

*Ray version and other system information (Python version, TensorFlow version, OS):*
 - ray version 0.8.7
 - python 3.6
 - tensorflow 2.3
 - Windows 10

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):
```
import ray
from ray import serve

import numpy as np
from typing import List

# BATCH SERVER
class BatchServer():
    def __init__(self):
        pass
    
    @serve.accept_batch
    def __call__(self, observation: List[np.array]) -> List[int]:
        output = [0 for i in range(len(observation))]
        return output

# ACTOR
@ray.remote
def deliver_inputs(handle):
    input_data = np.random.rand(3,3)
    results = handle.remote(observation=input_data)
    print(ray.get(results))

if __name__ == '__main__':
    serve.init()
    serve.create_backend("infer", BatchServer, config={"max_batch_size": 5})
    serve.create_endpoint("infer", backend="infer", route="/infer")
    init_handle = serve.get_handle("infer")
    
    deliver_inputs.remote(init_handle)
    
    serve.shutdown()
```
If we cannot run your script, we cannot fix your issue.

- [ X ] I have verified my script runs in a clean environment and reproduces the issue.
- [ X ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

**Comment by simon-mo** [comment]
Thanks for raising this issue @jasonyeoh87 

The underlying error seems to be
```
ray.exceptions.RayTaskError(TypeError): ray::RayServeWorker_BatchServer.handle_request() (pid=13009, ip=192.168.31.141)
  File "/Users/simonmo/Desktop/ray/ray/python/ray/serve/backend_worker.py", line 143, in wrap_to_ray_error
    raise exception
  File "/Users/simonmo/Desktop/ray/ray/python/ray/serve/backend_worker.py", line 288, in invoke_batch
    result_list = await call_method(*arg_list, **kwargs_list)
  File "/Users/simonmo/Desktop/ray/ray/python/ray/async_compat.py", line 31, in wrapper
    return func(*args, **kwargs)
TypeError: __call__() got multiple values for argument 'observation'
```

You can fix it by adding a `_unused_flask_request` for your `__call__` handler:
```diff
- def __call__(self, observation: List[np.array]) -> List[int]:
+ def __call__(self, _unused_flask_request, observation: List[np.array]) -> List[int]:
```

I understand this error is definitely not very clear and actionable and the API seems little hard to parse right now. Serve defaults the first argument to web requests. In the batch inference mode, the first argument is still injected with an empty object. 

We are working on improvement to this API and please let us know if you have any suggestions!

**Comment by jasonyeoh87** [comment]
Thank you for the help @simon-mo 
After making the fix you suggested (and now on latest ray version 0.9.0) I still have my worker dying unexpectedly with the following error message.

```
(pid=6996) E0820 12:56:54.874907  6996 18472 task_manager.cc:323] Task failed: IOError: cancelling all pending tasks of dead actor: Type=ACTOR_TASK, Language=PYTHON, Resources: {}, function_descriptor={type=PythonFunctionDescriptor, module_name=ray.serve.http_proxy, class_name=HTTPProxyActor, function_name=enqueue_request, function_hash=}, task_id=04974cbbc950b1067d75e23b0100, job_id=0100, num_args=6, num_returns=2, actor_task_spec={actor_id=7d75e23b0100, actor_caller_id=55c3b2b635949d81ffffffff0100, actor_counter=0}
2020-08-20 12:56:59,948	ERROR worker.py:1103 -- Possible unhandled error from worker: ray::__main__.deliver_inputs() (pid=6996, ip=172.20.10.4)
  File "python\ray\_raylet.pyx", line 477, in ray._raylet.execute_task
  File "C:\code\terrazero\testray2.py", line 22, in deliver_inputs
    print(ray.get(results))
  File "C:\Users\jason\Anaconda3\envs\py36\lib\site-packages\ray\worker.py", line 1558, in get
    raise value
ray.exceptions.RayActorError: The actor died unexpectedly before finishing this task.
```

The codes I have now look like this.
```
import ray
from ray import serve

import numpy as np
from typing import List

# BATCH SERVER
class BatchServer():
    def __init__(self):
        pass
    
    @serve.accept_batch
    def __call__(self, _unused_flask_request, observation: List[np.array]) -> List[int]:
        output = [0 for i in range(len(observation))]
        return output

# ACTOR
@ray.remote
def deliver_inputs(handle):
    input_data = np.random.rand(3,3)
    results = handle.remote(_unused_flask_request=None, observation=input_data)
    print(ray.get(results))

if __name__ == '__main__':
    serve.init()
    serve.create_backend("infer", BatchServer, config={"max_batch_size": 5})
    serve.create_endpoint("infer", backend="infer", route="/infer")
    init_handle = serve.get_handle("infer")
    
    deliver_inputs.remote(init_handle)
    serve.shutdown()
```

I also tried using a try/except block to catch the error happening in the Actor, but it does not seem to successfully catch anything. The actor process seems to die immediately after trying to use the Python handle I passed in.

**Comment by jasonyeoh87** [comment]
Hi @simon-mo 
I would like to add some findings.
I manage to solve the issues with some further tests, and found the issue to be caused by not calling one final ray.get() in the __main__ function before serve.shutdown(). 

Changing the final part to the following solves the issue.
```
    a = deliver_inputs.remote(init_handle)
    print(ray.get(a))   
    serve.shutdown()
 ```
